File: insurrection.js
/*
 * $Id$
 * Copyright 2004,2005 - Michael Sinz
 *
 * Some JavaScript configuration variables.
 * Set these to match your environment.
 *
 * Don't forget the matching insurrection.pl and insurrection.xml files.
 */

var Insurrection = new Object();

/*
 * This is the URL to the base of the repositories.
 * The form of the URL should be <base>/<repository>/<repofiles>
 *
 * For example, /svn/Web/trunk/... would be '/svn/' for the
 * base path, 'Web' for the repository name, and '/trunk/...' for
 * the files in the repository.
 */
Insurrection.Repositories_URL = '/svn/';

/*
 * This is the base URL for where the Insurrection CGIs live
 * This should include a leading and trailing '/' character.
 * (or just a single '/' if the CGIs are at the root.)
 */
Insurrection.CGI_URL = '/';

/*
 * Number of animation steps in the unfolding of the popups
 * To have things happen instantly then set this value to 1.
 */
Insurrection.SliderSteps = 12;

/*
 * Note - you should not need to change these unless you named
 * the CGIs differently from the default names.
 */

/*
 * This is the URL for the log.cgi
 */
Insurrection.log_CGI = Insurrection.CGI_URL + 'log.cgi';

/*
 * This is the URL for the get.cgi
 */
Insurrection.get_CGI = Insurrection.CGI_URL + 'get.cgi';

/*
 * This is the URL for the diff.cgi
 */
Insurrection.diff_CGI = Insurrection.CGI_URL + 'diff.cgi';

/*
 * This is the URL for the blame.cgi
 */
Insurrection.blame_CGI = Insurrection.CGI_URL + 'blame.cgi';

File: insurrection.pl
#
# $Id$
# Copyright 2004,2005 - Michael Sinz
#
# These are the constants that define some of the subversion configuration
# within the system plus some common code that is best to be shared.
#
# Don't forget the matching insurrection.js and insurrection.xsl files.
#

## This is the base of the repositories tree.  Repositories are within
our $SVN_BASE = '/home/subversion/repositories';

## This is the base of the authentication directory.  All of the authentication stuff lives here.
our $SVN_AUTH = '/home/subversion/authentication';

## This is the base of the logs directory.  All of the http logs live here.
our $SVN_LOGS = '/home/subversion/logs';

## This is where the svn binaries live.
our $SVN_BIN = '/home/subversion/svn/bin/';

## The domain for the EMail addresses...
our $EMAIL_DOMAIN = '@sinz.org';

## The default number of log entries to provide in the
## history.  If this is undef then we don't limit the
## entries.  (That would be bad)
our $SVN_LOG_ENTRIES = 20;  ## Default to the last 20 entries...

## The base path to the Subversion area on the server
## This path should start with a "/" and end with one.
## (Or, just be "/" which means the tree starts are root
our $SVN_URL_PATH = '/';

## The server relative path to the subversion repositories
## This path should start with "/" and end with one.
## Note that this should match the same setting in the
## insurrection.js file.
our $SVN_REPOSITORIES_URL = '/svn/';

## The specific binaries we use
our $SVN_CMD = $SVN_BIN . 'svn';
our $SVNLOOK_CMD = $SVN_BIN . 'svnlook';
our $SVNADMIN_CMD = $SVN_BIN . 'svnadmin';

1;

